File: backend/internal/model/milestone.go
package model

import "time"

type Milestone struct {
	MilestoneID uint       `gorm:"primaryKey" json:"milestoneId"`
	ProjectID   uint       `gorm:"index;not null" json:"projectId"`
	Title       string     `gorm:"not null" json:"title"`
	Description string     `json:"description"`
	DueDate     *time.Time `json:"dueDate,omitempty"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`
}

File: frontend/src/types/milestone.ts
export interface Milestone {
  milestoneId: number;
  projectId: number;
  title: string;
  description: string;
  dueDate?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMilestonePayload {
  title: string;
  description?: string;
  dueDate?: string;
}

export interface UpdateMilestonePayload {
  title?: string;
  description?: string;
  dueDate?: string;
}
